frontend/src/components/messages/AssistanceRequestTable.vue

```diff
@@ -0,0 +1,193 @@
+    },
+    markUnreadButtonInConversationThreadState: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  emits: ['openRequestConversation'],
+  data() {
+    return {
+      headers: [
+        { title: 'Status', align: 'start', key: 'status', sortable: true, width: '15%' },
+        { title: 'Subject', align: 'start', key: 'subject', sortable: true, width: '40%' },
+        { title: 'Topic', align: 'start', key: 'categoryName', sortable: true, width: '25%' },
+        { title: 'Last updated', align: 'start', key: 'lastConversationTime', sortable: true, width: '20%' },
+      ],
+      bodyCheckboxesSelected: [],
+      selectedRequestId: null,
+    }
+  },
+  computed: {
+    ...mapState(useMessagesStore, ['assistanceRequests']),
+  },
+  watch: {
+    markReadButtonState: {
+      handler() {
+        this.updateBodyCheckboxesReadUnread(true)
+      },
+    },
+    markUnreadButtonInMessageTableState: {
+      handler() {
+        this.updateBodyCheckboxesReadUnread(false)
+      },
+    },
+    markUnreadButtonInConversationThreadState: {
+      async handler() {
+        await this.updateMessageReadUnread(false, this.selectedRequestId)
+      },
+    },
+  },
+  methods: {
+    ...mapActions(useMessagesStore, ['updateAssistanceRequest']),
+
+    resetAllCheckboxes() {
+      this.bodyCheckboxesSelected = []
+    },
+    /**
+     * Update the body/item checkboxes to read or unread.
+     */
+    async updateBodyCheckboxesReadUnread(isRead) {
+      const selectedAssistanceRequestIds = this.bodyCheckboxesSelected
+      this.resetAllCheckboxes()
+      await Promise.all(
+        selectedAssistanceRequestIds.map(async (assistanceRequestId) => {
+          await this.updateMessageReadUnread(isRead, assistanceRequestId)
+        }),
+      )
+    },
+    /**
+     * Handles the row click event. When a row is clicked the assistanceRequests in context is displayed and marked as read.
+     */
+    async rowClickHandler(item, row) {
+      this.selectedRequestId = row?.item?.assistanceRequestId
+      this.$emit('openRequestConversation', this.selectedRequestId)
+      await this.updateMessageReadUnread(true, this.selectedRequestId)
+    },
+    async updateMessageReadUnread(isRead, assistanceRequestId) {
+      this.selectedAssistanceRequest = this.assistanceRequests?.find((item) => item.assistanceRequestId === assistanceRequestId)
+      let payload = {}
+      if (isRead) {
+        this.selectedAssistanceRequest.lastOpenedTime = new Date().toUTCString()
+        payload.lastOpenedTime = this.selectedAssistanceRequest.lastOpenedTime
+      }
+      if (this.selectedAssistanceRequest.isRead != isRead) {
+        this.selectedAssistanceRequest.isRead = isRead
+        payload.isRead = isRead
+      }
+      await this.updateAssistanceRequest(assistanceRequestId, payload)
+    },
+    isActionRequiredMessage(item) {
+      return item?.status === 'Action required'
+    },
+    getItemClass(item) {
+      let result = ''
```

Comment:
You might find it useful to do object style class binding instead: https://vuejs.org/guide/essentials/class-and-style.html rather than string concatentation.